slasher/beaconclient/validator_retrieval.go

```diff
@@ -0,0 +1,43 @@
+package beaconclient
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"go.opencensus.io/trace"
+)
+
+// RequestValidator requests validator public key from a beacon node via gRPC.
+func (bs *Service) RequestValidator(
```

Comment:
Although, the function doesn't always request from the beacon node, maybe we should call it `FindOrGetPublicKey`? Comments should detail it may also get it from the cache.

Reply:
replaced to FindOrGetPublicKeys